File: 백준/Gold/2023. 신기한 소수/신기한 소수.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.*;

public class Main {
        static BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        static BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        public static void main(String[] args) throws IOException {
                int n = Integer.parseInt(reader.readLine());
                for (int i = 2; i < 10; i++) {
                        if(checkPrimeNumber(i)) {
                                dfs(i, 1, n);
                        }
                }
                writer.flush();
                writer.close();
        }

        private static void dfs(int c, int length, int limit) throws IOException {
                if (length == limit) {
                        writer.write(c+"\n");
                        return;
                }

                for (int i = 1; i< 10;i++) {
                        int number = c * 10 + i;
                        if(checkPrimeNumber(number)) {
                                dfs(number, length+1, limit);
                        }
                }
        }

        private static boolean checkPrimeNumber(int num) {
                for(int i = 2; i <= num / 2; i++) {
                        if(num % i == 0) {
                                return false;
                        }
                }

                return true;
        }
}

File: 백준/Gold/2023. 신기한 소수/신기한 소수.py
import sys

sys.setrecursionlimit(100000)

input = sys.stdin.readline

def checkPrimeNumber(num):

    for i in range(2, num // 2 + 1):
        if num % i == 0:
            return False

    return True


def dfs(v, length, limit):
    if length == limit:
        print(v)
        return


    for i in range(1, 10):
        num = v * 10 + i
        if checkPrimeNumber(num):

            dfs(num, length+1, limit)


def main():
    n = int(input())

    for i in range(2, 10):
        if checkPrimeNumber(i):
            dfs(i, 1, n)

main()